# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../qtscriptgenerator/qtscriptgenerator-0.1.0-gcc44.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fix compiling with gcc >= 4.4.0

References:
http://code.google.com/p/qtscriptgenerator/issues/detail?id=37

diff -uNr qtscriptgenerator-src-0.1.0/generator.orig/parser/rpp/pp.h generator/parser/rpp/pp.h
--- qtscriptgenerator-src-0.1.0/generator.orig/parser/rpp/pp.h	2008-12-05 11:01:44.000000000 +0100
+++ qtscriptgenerator-src-0.1.0/generator/parser/rpp/pp.h	2008-12-13 10:42:22.000000000 +0100
@@ -30,6 +30,7 @@
 #include <cctype>
 
 #include <fcntl.h>
+#include <cstdio>
 
 #ifdef HAVE_MMAP
 #  include <sys/mman.h>
